lectures/python_essentials.md

```diff
@@ -940,14 +933,27 @@ Here's a solution:
 
 ```{code-cell} python3
 def f(seq_a, seq_b):
-    is_subset = True
     for a in seq_a:
         if a not in seq_b:
-            is_subset = False
-    return is_subset
+            return False
+    return True
 
 # == test == #
+print(f("ab", "cadb"))
+print(f("ab", "cjdb"))
+print(f([1, 2], [1, 2, 3]))
+print(f([1, 2, 3], [1, 2]))
+```
+
+An alternative, more pythonic solution using `all()`:
+
+```{code-cell} python3
+def f(seq_a, seq_b):
+  return all([i in seq_b for i in seq_a])
 
+  # == test == #
+print(f("ab", "cadb"))
+print(f("ab", "cjdb"))
 print(f([1, 2], [1, 2, 3]))
 print(f([1, 2, 3], [1, 2]))
 ```
```

Comment:
To show my changes to the exercise solution clearly, here is where I have rewritten.